contrib/STYLE.md

```diff
@@ -0,0 +1,208 @@
+
+Key Features of Black:
+
+    Consistency: Black enforces a single, consistent coding style across your project, eliminating style debates and allowing developers to focus on code logic.
+
+    Readability: By applying a standard formatting style, Black improves code readability, making it easier to understand and collaborate on projects.
+
+    Automation: Black automates the code formatting process, saving time and effort. It eliminates the need for manual formatting and reduces the likelihood of inconsistencies.
+
+### Git commit style
+
+Here’s a model Git commit message when contributing:
+```
+Summarize changes in around 50 characters or less
+
+More detailed explanatory text, if necessary. Wrap it to about 72
+characters or so. In some contexts, the first line is treated as the
+subject of the commit and the rest of the text as the body. The
+blank line separating the summary from the body is critical (unless
+you omit the body entirely); various tools like `log`, `shortlog`
+and `rebase` can get confused if you run the two together.
+
+Explain the problem that this commit is solving. Focus on why you
+are making this change as opposed to how (the code explains that).
+Are there side effects or other unintuitive consequences of this
+change? Here's the place to explain them.
+
+Further paragraphs come after blank lines.
+
+ - Bullet points are okay, too
+
+ - Typically a hyphen or asterisk is used for the bullet, preceded
+   by a single space, with blank lines in between, but conventions
+   vary here
+
+If you use an issue tracker, put references to them at the bottom,
+like this:
+
+Resolves: #123
+See also: #456, #789
+```
+
+
+## The six rules of a great commit.
+
+#### 1. Atomic Commits
+An “atomic” change revolves around one task or one fix.
+
+Atomic Approach
+ - Commit each fix or task as a separate change
+ - Only commit when a block of work is complete
+ - Commit each layout change separately
+ - Joint commit for layout file, code behind file, and additional resources
+
+Benefits
+
+- Easy to roll back without affecting other changes
+- Easy to make other changes on the fly
+- Easy to merge features to other branches
+
+*Caveat*: When working with new features, an atomic commit will often consist of multiple files, since a layout file, code behind file, and additional resources may have been added/modified. You don’t want to commit all of these separately, because if you had to roll back the application to a state before the feature was added, it would involve multiple commit entries, and that can get confusing
+
+#### 2. Separate subject from body with a blank line
```

Comment:
The rules here should be part of the pull request task checklist. There are many commits with the message 'fix' 'fix2' 'fix3' in this repository, with little squashing or rebasing. `--fixup` and interactive rebases should also be mentioned.

Reply:
See: ebeec8e3c9a1bae6cab12fd7dd83d5b6ae6d3d82